zerver/lib/actions.py

```diff
@@ -2772,16 +2795,11 @@ def check_update_message(
     # 2. This is a topic-only edit for a (no topic) message, OR:
     # 3. This is a topic-only edit and you are an admin, OR:
     # 4. This is a topic-only edit and your realm allows users to edit topics.
-    if message.sender == user_profile:
```

Comment:
I think these comments should come along to the new function, perhaps moving each before one of the conditionals.

Also, there's a typo in the commit message.

Reply:
Fixed this.